### ISSUE TITLE
maxele.63 [adcirc to netcdf] segmentation error

### ISSUE THREAD

**Issue by ArslaanK** [jasonfleming/asgs#28]
Hi, 
 I tried with multiple maxele.63 files (different sources) but everytime it is yielding this error, can you look into the matter please?
thanks
**arslaan@Linux:~/Documents/KalpanaTesting_DONE/script$ *adcirc2netcdf.x* --meshfile 'fort.14' --attfile 'generic_atts.txt' --datafile 'maxele.63'**
INFO: adcirc2netcdf was compiled with the following netcdf library: 4.4.0 of Mar 29 2016 11:41:40 $
INFO: Processing --meshfile fort.14.
INFO: Processing --attfile generic_atts.txt.
INFO: Processing --datafile maxele.63.
INFO: adcirc2netcdf.f90: Checking number of nodes in data file.
INFO: adcirc2netcdf.f90: Searching for file maxele.63 ...
INFO: adcirc2netcdf.f90: The file maxele.63 was found.
INFO: adcirc2netcdf.f90: The file maxele.63 was opened successfully.
INFO: Creating NetCDF file 'maxele.63.nc'.
INFO: adcirc2netcdf.f90: Opening netcdf metadata/attributes file.
INFO: adcirc2netcdf.f90: Searching for file generic_atts.txt ...
INFO: adcirc2netcdf.f90: The file generic_atts.txt was found.
INFO: adcirc2netcdf.f90: The file generic_atts.txt was opened successfully.
INFO: Finished reading metadata/attributes file.
INFO: adcirc2netcdf.f90: Searching for file fort.14 ...
INFO: adcirc2netcdf.f90: The file fort.14 was found.
INFO: adcirc2netcdf.f90: The file fort.14 was opened successfully.
INFO: Mesh file comment line: FLAT88_fluxbndbathycorrected.grd
INFO: Reading mesh file dimensions.
INFO: Allocating memory for elevation specified boundaries.
INFO: Allocating memory for flux specified boundaries.
WARNING: Number of flux boundary nodes was set to 583 but 582 were found.
INFO: Finished reading mesh file dimensions.
INFO: Reading mesh file coordinates, connectivity, and boundary data.
INFO: adcirc2netcdf.f90: Searching for file fort.14 ...
INFO: adcirc2netcdf.f90: The file fort.14 was found.
INFO: adcirc2netcdf.f90: The file fort.14 was opened successfully.
INFO: Finished reading mesh file coordinates, connectivity, and boundary data.
INFO: Writing mesh definitions to netcdf.
INFO: Finished writing mesh definitions to netcdf.
 **INFO: Adding data attributes to netCDF file.
 fn%numVarNetCDF=           0
 nc_dimid =           1           0
 initFileMetaData : enter
 num xdmf=           1
 xdmf init finished
 netcdf vartype init finished
 netcdf init(1) finished**
 initFileMetaData : return
 INFO: Finished adding data attributes to netCDF file.
INFO: Mesh has been written to the netCDF file.
INFO: adcirc2netcdf.f90: Searching for file maxele.63 ...
INFO: adcirc2netcdf.f90: The file maxele.63 was found.
INFO: adcirc2netcdf.f90: The file maxele.63 was opened successfully.
 
**allocating f%rdata**

**Program received signal SIGSEGV: Segmentation fault - invalid memory reference.**

**Backtrace for this error:**
**#0  0x7FA72A45BE08**
**#1  0x7FA72A45AF90**
**#2  0x7FA729DA24AF**
**#3  0x7FA72A77F060**
**#4  0x7FA72A7B37B1**
**#5  0x414C11 in __asgsio_MOD_writeonedataset**
**#6  0x403ED8 in MAIN__ at adcirc2netcdf.f90:?**
**Segmentation fault (core dumped)**

**Comment by zcobell** [comment]
Which type of compiler are you using? I've encountered this error with Intel on certain machines, but it appears to be a compiler issue an not necessarily a code issue. The fix with the Intel compilers on one of my machines was accomplished by the addition of the compiler flag `-heap-arrays 8192` where 8192 is the machine's stack size (`ulimit -s`). This seems to clear up the issue because Intel does not automatically move implicitly created arrays to the heap, leading to a segfault. These implicit arrays are created inside the netCDF routines. The fix needed to be applied to both the netCDF Fortran library and the `adcirc2netcdf.f90` code itself.

**Comment by ArslaanK** [comment]
I am using **gfortran** compiler , I tried compiling with intel but it gives me a error and I couldn't figure out yet.

here is the error

**:~/Documents/KalpanaTesting_DONE/script/asgs/output$ make adcirc2netcdf compiler=intel MACHINENAME=poseidon NETCDF=enable NETCDF4=enable NETCDF4_COMPRESSION=enable
ifort -cpp -i-dynamic -DASGSNETCDF -DASGSNETCDF -DHAVE_NETCDF4 -DNETCDF_CAN_DEFLATE -I. -I/home/fhrl/Documents/Surge/scripts/netcdf/include  -L/home/fhrl/Documents/Surge/scripts/netcdf/lib -c logging.f90  -lnetcdff
ifort: command line remark #10148: option '-i-dynamic' not supported
logging.f90(222): error #7013: This module file was not generated by any release of this compiler.   [NETCDF]
use netcdf
----^
logging.f90(420): error #7013: This module file was not generated by any release of this compiler.   [NETCDF]
use netcdf
----^
logging.f90(425): error #6404: This name does not have a type, and must have an explicit type.   [NF90_NOERR]
if(ncstatus.ne.nf90_noerr)then
---------------^
logging.f90(426): error #6404: This name does not have a type, and must have an explicit type.   [NF90_STRERROR]
   write(*,'(a,a)') "ERROR: ",trim(nf90_strerror(ncstatus))
-----------------------------------^
logging.f90(426): error #6362: The data types of the argument(s) are invalid.   [TRIM]
   write(*,'(a,a)') "ERROR: ",trim(nf90_strerror(ncstatus))
-----------------------------------^
compilation aborted for logging.f90 (code 1)
makefile:170: recipe for target 'logging.mod' failed
make: *** [logging.mod] Error 1**

**Comment by zcobell** [comment]
Unfortunately, I'm not familiar enough with `gfortran` to know if it has some of the same heap allocation issues as Intel, though I don't think it does. The error from your intel compile is because your netCDF library was compiled with gfortran. You may try to debug your original error using `gdb`, which is the GNU debugger. It should give you some additional information that can help you see exactly where the code is going wrong. You should include `-g -fbacktrace` when compiling both netCDF and adcirc2netcdf to generate debugging information that can be used by `gdb`.

**Comment by ArslaanK** [comment]
Thanks Zach 
I will try to recompile fortran library with intel and will check if the problem persists

**Comment by jasonfleming** [comment]
Hello @ArslaanK I concur with @zcobell about the intel compile. On the issue of the segmentation fault, I think I found and fixed this a while back in the 2014stable branch. I just tried to reproduce this issue with some test files from the shinnecock inlet test problem, and everything went as expected. I will close this issue, but we can reopen it if things are still not running as expected for you.
